fleetspeak/src/config/client/config.go

```diff
@@ -0,0 +1,100 @@
+	cpb "github.com/google/fleetspeak/fleetspeak/src/config/proto/fleetspeak_config"
+)
+
+func WriteLinuxConfig(cfg cpb.Config, trustedPEM []byte) error {
+	if cfg.LinuxClientConfigurationFile == "" {
+		return nil
+	}
+	out := gpb.Config{
+		TrustedCerts: string(trustedPEM),
+		Server:       cfg.PublicHostPort,
+		ClientLabel:  []string{cfg.ComponentsConfig.RequiredLabel},
+		PersistenceHandler: &gpb.Config_FilesystemHandler{
+			FilesystemHandler: &gpb.FilesystemHandler{
+				ConfigurationDirectory: "/etc/fleetspeak-client",
+				StateFile:              "/var/lib/misc/fleetspeak-client.state",
+			}},
+		Streaming: !cfg.ComponentsConfig.HttpsConfig.DisableStreaming,
+	}
+
+	f, err := os.OpenFile(cfg.LinuxClientConfigurationFile, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0644)
+	if err != nil {
+		return fmt.Errorf("unable to open server linux client configuration file [%s] for writing: %v", cfg.LinuxClientConfigurationFile, err)
+	}
+	if err := proto.MarshalText(f, &out); err != nil {
+		return fmt.Errorf("failed to write linux client configuration file [%s]: %v", cfg.LinuxClientConfigurationFile, err)
+	}
+	if err := f.Close(); err != nil {
+		return fmt.Errorf("failed to write linux client configuration file [%s]: %v", cfg.LinuxClientConfigurationFile, err)
+	}
+
+	return nil
+}
+
+func WriteDarwinConfig(cfg cpb.Config, trustedPEM []byte) error {
+	if cfg.DarwinClientConfigurationFile == "" {
+		return nil
+	}
+	out := gpb.Config{
+		TrustedCerts: string(trustedPEM),
+		Server:       cfg.PublicHostPort,
+		ClientLabel:  []string{cfg.ComponentsConfig.RequiredLabel},
+		PersistenceHandler: &gpb.Config_FilesystemHandler{
+			FilesystemHandler: &gpb.FilesystemHandler{
+				ConfigurationDirectory: "/etc/fleetspeak-client",
+				StateFile:              "/etc/fleetspeak-client/state",
+			}},
+		Streaming: !cfg.ComponentsConfig.HttpsConfig.DisableStreaming,
+	}
+
+	f, err := os.OpenFile(cfg.DarwinClientConfigurationFile, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0644)
+	if err != nil {
+		return fmt.Errorf("unable to open server darwin client configuration file [%s] for writing: %v", cfg.DarwinClientConfigurationFile, err)
+	}
+	if err := proto.MarshalText(f, &out); err != nil {
+		return fmt.Errorf("failed to write darwin client configuration file [%s]: %v", cfg.DarwinClientConfigurationFile, err)
+	}
+	if err := f.Close(); err != nil {
+		return fmt.Errorf("failed to write darwin client configuration file [%s]: %v", cfg.DarwinClientConfigurationFile, err)
+	}
+
+	return nil
+}
+
+func WriteWindowsConfig(cfg cpb.Config, trustedPEM []byte) error {
+	if cfg.WindowsClientConfigurationFile == "" {
+		return nil
+	}
+	out := gpb.Config{
+		TrustedCerts: string(trustedPEM),
+		Server:       cfg.PublicHostPort,
+		ClientLabel:  []string{cfg.ComponentsConfig.RequiredLabel},
+		PersistenceHandler: &gpb.Config_RegistryHandler{
+			RegisteryHandler: &gpb.RegistryHandler{
+				ConfigurationKey: `HKEY_LOCAL_MACHINE\SOFTWARE\FleetspeakClient`,
```

Comment:
GRR currently assumes this to be HKEY_LOCAL_MACHINE\SOFTWARE\Fleetspeak by default.

The 'Client' suffix is unnecessary IMO since only clients will use the registry, not servers as well.